core/src/impl/Kokkos_Atomic_View.hpp

```diff
@@ -335,6 +335,8 @@ class AtomicDataElement {
   }
 
   KOKKOS_INLINE_FUNCTION
+  // FIXME volatile-qualified return type  is deprecated - safe to simply remove
+  // volatile on return type here?
   operator volatile non_const_value_type() volatile const {
```

Comment:
@masterleinad @nliber thanks for the feedback, do you have any suggestions on the volatile return type here (possibly this is addressed via some updates in PR #3719)? Let me know if this should be resolved as part of this PR, or if I should clean up stray comments and get the changes in this PR as is

Reply:
I haven't looked into the `volatile` warnings in #3719.

Reply:
This is the last FIXME and since it lives in the `Impl` namespace, it should be fine to change the return type if that passes CI. I would change it here.

Reply:
Thanks for suggestion, pushed the change with the volatile type removed from the return type